packages/dapp/src/const/MenuConfig/useWalletSelectContent.tsx

```diff
@@ -34,43 +50,68 @@ export const useWalletSelectContent = () => {
     [connect, onWalletConnect],
   );
 
+  useEffect(() => {
+    verifyMultisigEnvironment();
+  }, [account?.address]);
+
   const walletMenuItems = useMemo<MenuListItem[]>(() => {
-    const installedWallets = supportedWallets.filter((wallet) =>
-      wallet.installed(),
-    );
+    let availableWallets: Wallet[] = [];
 
-    const notInstalledWallets = supportedWallets.filter(
-      (wallet) => !wallet.installed() && wallet.name !== 'Default Wallet', // always remove Default Wallet from not installed Wallets
-    );
-    const _output = [...installedWallets, ...notInstalledWallets].map(
-      (wallet) => {
-        return {
-          label: wallet.name,
-          prefixIcon: (
-            <Avatar
-              src={wallet.icon}
-              alt={`${wallet.name}-wallet-logo`}
-              sx={{ height: '32px', width: '32px' }}
-            />
-          ),
-          showMoreIcon: false,
-          onClick: () => {
-            login(wallet);
-            onCloseAllNavbarMenus();
-            onWelcomeScreenEntered(true);
-            trackEvent({
-              category: TrackingCategories.Wallet,
-              action: TrackingActions.ChooseWallet,
-              label: `choose-wallet-${wallet}`,
-              data: { usedWallet: wallet.name },
-              disableTrackingTool: [EventTrackingTools.arcx],
-            });
-          },
-        };
+    Promise.all(supportedWallets.map((wallet) => wallet.installed())).then(
+      (installed) => {
+        // separate into installed and not installed wallets
+        const installedWallets = supportedWallets.filter(
+          (_, index) => installed[index],
+        );
+        // always remove Default Wallet from not installed Wallets
+        const notInstalledWallets = supportedWallets.filter(
+          (wallet, index) =>
+            !installed[index] && wallet.name !== 'Default Wallet',
+        );
+        availableWallets = [...installedWallets, ...notInstalledWallets];
```

Comment:
Are you sure this works fine inside useMemo? We will exit the function before `availableWallets` can be set because Promise.all is async. useEffect maybe?